juju/juju
# InstanceMutater should handle different required profiles depending on broker

## Description of change

There is a different list of lxd profiles for juju lxd machines and juju lxd containers.  Both use "default", but only a juju lxd machine uses the "juju-<modelname>" profile.

Model cache WatchMachines() should not report containers.

Set the machine's modification status to Idle before making any changes to the profiles.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=instance-mutater
2. juju bootstrap localhost
3. juju deploy ./testcharms/charm-repo/quantal/lxd-profile-alt --to lxd -n 2
2. wait for both units to be active
3. juju upgrade-charm lxd-profile-alt  --path ./testcharms/charm-repo/quantal/lxd-profile-alt
4. verify that both are successful and that their profiles have been updated
